### ISSUE TITLE
FastList cannot be cast to java.util.ArrayList error

### ISSUE THREAD

**Issue by gogainda** [mvel/mvel#168]
I noticed that if I am using ```setDefaultOptimizer(DYNAMIC);``` the program could fail with an error
```
java.lang.ClassCastException: org.mvel2.util.FastList cannot be cast to java.util.ArrayList
	at ASMAccessorImpl_103198053115394498196380.getValue(Unknown Source)
	at org.mvel2.optimizers.dynamic.DynamicGetAccessor.getValue(DynamicGetAccessor.java:78)
	at org.mvel2.ast.Union.getReducedValueAccelerated(Union.java:39)
	at org.mvel2.MVELRuntime.execute(MVELRuntime.java:85)
	at org.mvel2.compiler.CompiledExpression.getDirectValue(CompiledExpression.java:123)
	at org.mvel2.compiler.CompiledExpression.getValue(CompiledExpression.java:119)
	at org.mvel2.MVEL.executeExpression(MVEL.java:929)
	at com.mvel.RulesValidatorCompiled.isValid(RulesValidatorCompiled.java:22)
	at com.mvel.App.main(App.java:35)
Exception in thread "main" java.lang.RuntimeException: RecordImpl{digest='ui4n002cgu3eus3kb8g9dn1uqj'}
	at com.mvel.App.main(App.java:39)
```
but the same code works with ```setDefaultOptimizer(SAFE_REFLECTIVE);```
I attached the example project. If you try to run it several times it fails at exactly same spot, on 101st step
```
been executed times:101
```
[mvelpoc.zip](https://github.com/mvel/mvel/files/2475659/mvelpoc.zip)

**Comment by sanjay9977** [comment]
Can you past your Expression here..

**Comment by gogainda** [comment]
Sure, ```add(new ExpressionCompiler("[\"001\", \"002\", \"003\", \"005\", \"007\", \"008\", \"009\"].contains(digest)").compile());```. 
Basically this expression is compiled only once and run multiple times with random ```digest```

**Comment by MalcolmOdd** [comment]
This looks as a duplicate of [#113](https://github.com/mvel/mvel/issues/113)
